pkg/webhook/inject/inject.go

```diff
@@ -34,25 +33,44 @@ import (
 	genericnamespace "github.com/chaos-mesh/chaos-mesh/pkg/selector/generic/namespace"
 	podselector "github.com/chaos-mesh/chaos-mesh/pkg/selector/pod"
 	"github.com/chaos-mesh/chaos-mesh/pkg/webhook/config"
+	"github.com/go-logr/logr"
 )
 
-var log = ctrl.Log.WithName("inject-webhook")
-
 var ignoredNamespaces = []string{
 	metav1.NamespaceSystem,
 	metav1.NamespacePublic,
 }
 
+type Injector struct {
+	logger        logr.Logger
+	kubeclient    client.Client
+	cfg           *config.Config
+	ControllerCfg *controllerCfg.ChaosControllerConfig
+	Metrics       *metrics.ChaosControllerManagerMetricsCollector
+}
+
 const (
 	// StatusInjected is the annotation value for /status that indicates an injection was already performed on this pod
 	StatusInjected = "injected"
 )
 
+func NewInjector(kubeclient client.Client, cfg *config.Config, controllerCfg *controllerCfg.ChaosControllerConfig, metrics *metrics.ChaosControllerManagerMetricsCollector) *Injector {
+	return &Injector{
+		kubeclient:    kubeclient,
+		cfg:           cfg,
+		ControllerCfg: controllerCfg,
+		Metrics:       metrics,
+	}
+}
+
 // Inject do pod template config inject
-func Inject(res *v1.AdmissionRequest, cli client.Client, cfg *config.Config, controllerCfg *controllerCfg.ChaosControllerConfig, metrics *metrics.ChaosControllerManagerMetricsCollector) *v1.AdmissionResponse {
+func (i Injector) Inject(res *v1.AdmissionRequest, cli client.Client, cfg *config.Config, controllerCfg *controllerCfg.ChaosControllerConfig, metrics *metrics.ChaosControllerManagerMetricsCollector) *v1.AdmissionResponse {
```

Comment:
please change the signature of method `Inject`, as `Injector` already contains fields with type `client.Cient`, `config.Config`. You could use them instead of acquiring new methods parameters.